pymc/distributions/truncated.py

```diff
@@ -209,17 +227,21 @@ def rv_op(cls, dist, lower, upper, max_n_steps, size=None):
             pass
 
         # Fallback to rejection sampling
-        def loop_fn(truncated_rv, reject_draws, lower, upper, rng, *rv_inputs):
-            next_rng, new_truncated_rv = dist.owner.op.make_node(rng, *rv_inputs).outputs
-            truncated_rv = pt.set_subtensor(
-                truncated_rv[reject_draws],
-                new_truncated_rv[reject_draws],
-            )
+        def loop_fn(truncated_rv, reject_draws, lower, upper, *rv_inputs):
+            new_truncated_rv = dist.owner.clone_with_new_inputs(rv_inputs).outputs[dist_out_idx]
+            # Avoid scalar boolean indexing
+            if truncated_rv.type.ndim == 0:
+                truncated_rv = new_truncated_rv
+            else:
+                truncated_rv = pt.set_subtensor(
+                    truncated_rv[reject_draws],
+                    new_truncated_rv[reject_draws],
+                )
             reject_draws = pt.or_((truncated_rv < lower), (truncated_rv > upper))
```

Comment:
Shouldn't this also have some kind of `pt.and_(not(reject_draws), ...` so that the draws that were already accepted don't get resampled? I fail to see where that is happening.

Reply:
The draws that were already accepted will always be within upper and lower. The set_subtensor only changes the indexes that were not already valid.